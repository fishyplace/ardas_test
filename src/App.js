import React, { Component } from 'react';
import './App.css';
import * as data from './assets/tasks.json';
import ExtendedList from './components/ExtendedList/ExtendedList';
import ElementEditor from './containers/ElementEditor/ElementEditor';
import { Switch, Route } from 'react-router-dom';

class App extends Component {

  state = {
    data: data.default
  }

  editEntryHandler = (id, newName) => {
    const entries = [...this.state.data];

    for(let i = 0; i < entries.length; i++) if (entries[i].id === +id) entries[i] = {...entries[i], name:newName};

    this.setState({data: entries});
  }

  render() {

    let input = null;
    if (this.state.data.length > 0) {
      input = this.state.data.filter(el => el.obj_status === 'active');
    }


    return (
      <Switch>
        <Route exact path="/" render={() => <ExtendedList data={input} />} />
        <Route path="/edit/:id" render={props => <ElementEditor {...props} data={input} editEntryHandler={this.editEntryHandler.bind(this)} />} />
      </Switch>
    );
  }
}

export default App;
