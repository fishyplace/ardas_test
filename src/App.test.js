import React from 'react';
import App from './App';

import { shallow } from 'enzyme';

it('should render without crashing', () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});
