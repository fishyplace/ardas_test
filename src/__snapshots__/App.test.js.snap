// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    path="/edit/:id"
    render={[Function]}
  />
</Switch>
`;
